test/distributed/_tensor/test_random_ops.py

```diff
@@ -142,6 +142,49 @@ def test_deterministic_dropout_1d(self):
                     local_tensor[other_slice, :],
                 )
 
+    @with_comms
+    @skip_unless_torch_gpu
+    def test_deterministic_rand_1d(self):
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        size = [4, 4 * self.world_size]
+
+        dtensor = torch.distributed._tensor.rand(
+            size, device_mesh=device_mesh, placements=[Shard(1)]
+        )
+        local_tensor = funcol.all_gather_tensor(
+            dtensor.to_local(), gather_dim=0, group=(device_mesh, 0)
+        )
+
+        # compare with local tensors from other ranks
+        self_slice = slice(4 * self.rank, 4 * self.rank + 4)
+        for other_rank in range(self.world_size):
+            if self.rank != other_rank:
+                # other rank should have an identical local tensor
+                other_slice = slice(4 * other_rank, 4 * other_rank + 4)
+                self.assertNotEqual(
+                    local_tensor[self_slice, :],
+                    local_tensor[other_slice, :],
+                )
+
+        torch.cuda.manual_seed(self.rank)
```

Comment:
It would be nice to mention that even if different seeds are set, we can still guarantee same value for Replicate DTensor.